app/models/manageiq/providers/openshift/container_manager/container_image.rb

```diff
@@ -1,10 +1,48 @@
 class ManageIQ::Providers::Openshift::ContainerManager::ContainerImage < ContainerImage
-  def annotate_deny_execution(causing_policy)
+  def annotate_image(annotations)
     ext_management_system.annotate(
       "image",
       digest,
+      annotations
+    )
+  end
+
+  def openscap_summary
+    failed_rules = openscap_rule_results.where(:result => "fail").group(:severity).count
+    [[['High'], 'Critical', 3],
+     [['Medium'], 'Important', 2],
+     [['Low'], 'Medium', 1],
+     [['Info', 'Unknown'], 'Low', 0]].collect do |severities, label, index|
+      {
+        :label         => label,
+        :severityIndex => index,
+        :data          => failed_rules.select { |sev| severities.include?(sev) }.values.sum
+      }
+    end
+  end
+
+  def security_quality_annotation(compliant)
+    {"quality.images.openshift.io/vulnerability.manageiq" => {
+      :name        => "ManageIQ",
+      :timestamp   => Time.now.utc.to_i,
+      :description => "OpenSCAP Score",
+      :reference   => "",
+      :compliant   => compliant,
+      :summary     => openscap_summary
+    }.to_json}
+  end
+
+  def annotate_allow_execution(causing_policy)
+    annotate_image({
+      "security.manageiq.org/successful-policy" => causing_policy,
+      "images.openshift.io/allow-execution"     => "true"
+    }.merge!(security_quality_annotation(true)))
+  end
+
+  def annotate_deny_execution(causing_policy)
+    annotate_image({
       "security.manageiq.org/failed-policy" => causing_policy,
       "images.openshift.io/deny-execution"  => "true"
```

Comment:
[out of scope for this PR]
This ties making the scan details annotation with making allow/deny-execution annotation.
Until now the way user decided whether to make `deny-execution` annotations is by using or not the action on a policy reacting to scan complete event.
Now it might become useful to make policy annotation without denying execution, or have extra conditions for denying.
Also, conceivably one might want to deny/allow in other policies that don't have an OpenSCAP scan at hand.
=> in future, it may be worth splitting this, making scan details annotation immediately on scan (or as independent action?), and having actions that make only deny/allow annotation...